api/src/main/java/org/pmiops/workbench/db/model/DbUser.java

```diff
@@ -695,17 +697,56 @@ public void setProfessionalUrl(String professionalUrl) {
     this.professionalUrl = professionalUrl;
   }
 
-  @OneToOne(
+  /**
+   * The mapping from user:address is logically 1:1 in the app, but is modeled as 1:many in the DB.
+   * As a workaround until we're ready to correrct the constraint in the schema, I've added a
+   * binding to a Set to allow multiple addresses to be stored and also exposed a transient setter
+   * and getter for a single address.
+   *
+   * @return
+   */
+  @OneToMany(
       cascade = CascadeType.ALL,
       orphanRemoval = true,
       fetch = FetchType.LAZY,
       mappedBy = "user")
-  public DbAddress getAddress() {
-    return address;
+  public Set<DbAddress> getAddresses() {
+    return addresses;
+  }
+
+  public void setAddresses(Set<DbAddress> address) {
+    this.addresses = address;
   }
 
+  /**
+   * This set may be initially null, but if this method is called with a null parameter, we set it
+   * to empty.
+   *
+   * @param address
+   */
+  @Transient
+  @Nullable
   public void setAddress(DbAddress address) {
-    this.address = address;
+    if (address != null) {
```

Comment:
Heads up: this line is causing unit test errors, seems like due to https://stackoverflow.com/questions/7428089/unsupportedoperationexception-merge-saving-many-to-many-relation-with-hibernate since the Collections methods create immutable sets.

This seems to work:

```suggestion
addresses.clear();
if (address != null) {
  addresses.add(address);
}
```

Reply:
I did find that, and locally I have

```
  public void setAddresses(Set<DbAddress> addresses) {
    this.addresses.clear(); // keep same collection  object
    addresses.forEach(a -> a.setUser(this)); // not sure about this
    this.addresses.addAll(addresses);
  }

  @Transient
  public void setAddress(@Nullable DbAddress address) {
    setAddresses(Optional.ofNullable(address)
        .map(Collections::singleton) // this set is not persisted; used to pass the address
        .orElse(Collections.emptySet()));
  }
```